velox/dwio/parquet/tests/ParquetTpchTest.cpp

```diff
@@ -0,0 +1,306 @@
+      int64_t numSplits) const {
+    LocalReadFile lfs(filePath);
+    const int fileSize = lfs.size();
+    const int splitSize = fileSize / numSplits;
+    std::vector<std::shared_ptr<connector::hive::HiveConnectorSplit>> splits;
+
+    // Add all the splits except the last one.
+    int i = 0;
+    for (; i < numSplits - 1; i++) {
+      auto split = makeHiveConnectorSplit(filePath, i * splitSize, splitSize);
+      split->fileFormat = dwio::common::FileFormat::PARQUET;
+      splits.push_back(std::move(split));
+    }
+    // Add the last split with the remaining as start.
+    auto split = makeHiveConnectorSplit(filePath, i * splitSize);
+    split->fileFormat = dwio::common::FileFormat::PARQUET;
+    splits.push_back(std::move(split));
+    return splits;
+  }
+
+  void addSplitsToTask(
+      exec::Task* task,
+      bool& noMoreSplits,
+      const std::vector<std::string>& filePaths,
+      int sourcePlanNodeId = 0,
+      int64_t numSplits = 10) const {
+    if (!noMoreSplits) {
+      for (const auto& filePath : filePaths) {
+        auto const& splits = makeSplits(filePath, numSplits);
+        for (const auto& split : splits) {
+          task->addSplit(std::to_string(sourcePlanNodeId), exec::Split(split));
+        }
+      }
+      task->noMoreSplits(std::to_string(sourcePlanNodeId));
+      noMoreSplits = true;
+    }
+  }
+
+  // Write the Lineitem TPCH table to a Parquet file and return the file
+  // location.
+  std::string writeDuckDBTPCHLineitemTableToParquet() const {
+    constexpr std::string_view tableName("lineitem");
+    // Lineitem SF=0.01 has 60175 rows
+    // Set the number of rows in a RowGroup so that the generated file contains
+    // multiple RowGroups.
+    constexpr int rowGroupNumRows = 15'000;
+    const auto& filePath =
+        fmt::format("{}/{}.parquet", tempDirectory_->path, tableName);
+    // Convert decimal columns to double.
+    const auto& query = fmt::format(
+        "COPY (SELECT l_orderkey as orderkey, l_partkey as partkey, l_suppkey as suppkey, l_linenumber as linenumber, "
+        "l_quantity::DOUBLE as quantity, l_extendedprice::DOUBLE as extendedprice, l_discount::DOUBLE as discount, "
+        "l_tax::DOUBLE as tax, l_returnflag as returnflag, l_linestatus as linestatus, "
+        "l_shipdate as shipdate, l_receiptdate as receiptdate, "
+        "l_shipinstruct as shipinstruct, l_shipmode as shipmode, l_comment as comment "
+        "FROM {}) TO '{}' (FORMAT 'parquet', ROW_GROUP_SIZE {})",
+        tableName,
+        filePath,
+        rowGroupNumRows);
+    duckDb_->execute(query);
+    return filePath;
+  }
+
+  static std::shared_ptr<DuckDbQueryRunner> duckDb_;
+  std::shared_ptr<exec::test::TempDirectoryPath> tempDirectory_;
+};
+
+std::shared_ptr<DuckDbQueryRunner> ParquetTpchTest::duckDb_ = nullptr;
+
+TEST_F(ParquetTpchTest, tpchQ1) {
+  const auto& lineitemFilePath = writeDuckDBTPCHLineitemTableToParquet();
+
+  // TPCH Q1
+  auto rowType = ROW(
```

Comment:
perhaps, add a member variable with the lineitem schema, then add a helper method to get types for selected columns, e.g.

```
auto rowType = getLineitemColumns({"returnflag",
       "linestatus",
       "quantity",
       "extendedprice",
       "discount",
       "tax",
       "shipdate"});
```